handlers/justcall/justcall.go

```diff
@@ -0,0 +1,239 @@
+		}
+		date = date.UTC()
+	}
+
+	urn, err := handlers.StrictTelForCountry(payload.Data.From, c.Country())
+	if err != nil {
+		return nil, handlers.WriteAndLogRequestError(ctx, h, c, w, r, err)
+	}
+
+	// build our msg
+	msg := h.Backend().NewIncomingMsg(c, urn, payload.Data.Content, clog).WithExternalID(fmt.Sprint(payload.Data.MessageID)).WithReceivedOn(date)
+
+	if len(payload.Data.MMS) > 0 {
+		msg.WithAttachment(payload.Data.MMS[0].MediaURL)
+	}
+
+	// and finally write our message
+	return handlers.WriteMsgsAndResponse(ctx, h, []courier.Msg{msg}, w, r, clog)
+}
+
+var statusMapping = map[string]courier.MsgStatusValue{
+	"delivered":   courier.MsgDelivered,
+	"sent":        courier.MsgSent,
+	"undelivered": courier.MsgErrored,
+	"failed":      courier.MsgFailed,
+}
+
+func (h *handler) statusMessage(ctx context.Context, c courier.Channel, w http.ResponseWriter, r *http.Request, clog *courier.ChannelLog) ([]courier.Event, error) {
+	payload := &moPayload{}
+	err := handlers.DecodeAndValidateJSON(payload, r)
+	if err != nil {
+		return nil, handlers.WriteAndLogRequestError(ctx, h, c, w, r, err)
+	}
+
+	if payload.Data.Type != "sms" || payload.Data.Direction != "O" {
+		return nil, handlers.WriteAndLogRequestIgnored(ctx, h, c, w, r, "Ignoring request, no message")
+	}
+
+	msgStatus, found := statusMapping[payload.Data.Status]
+	if !found {
+		return nil, handlers.WriteAndLogRequestError(ctx, h, c, w, r, fmt.Errorf("unknown status '%s', must be one of send, delivered, undelivered, failed", payload.Data.Status))
+	}
+	// write our status
+	status := h.Backend().NewMsgStatusForExternalID(c, fmt.Sprint(payload.Data.MessageID), msgStatus, clog)
+	return handlers.WriteMsgStatusAndResponse(ctx, h, c, status, w, r)
+}
+
+type mtPayload struct {
+	From     string `json:"from"`
+	To       string `json:"to"`
+	Body     string `json:"body"`
+	MediaURL string `json:"media_url,omitempty"`
+}
+
+// Send implements courier.ChannelHandler
+func (h *handler) Send(ctx context.Context, msg courier.Msg, clog *courier.ChannelLog) (courier.MsgStatus, error) {
+	apiKey := msg.Channel().StringConfigForKey(courier.ConfigAPIKey, "")
+	if apiKey == "" {
+		return nil, fmt.Errorf("no API key set for JCL channel")
+	}
+
+	apiSecret := msg.Channel().StringConfigForKey(courier.ConfigSecret, "")
+	if apiSecret == "" {
+		return nil, fmt.Errorf("no API secret set for JCL channel")
+	}
+
+	status := h.Backend().NewMsgStatusForID(msg.Channel(), msg.ID(), courier.MsgErrored, clog)
+	mediaURLs := make([]string, 0, 5)
+	text := msg.Text()
+
+	if len(msg.Attachments()) <= 5 {
+		for _, a := range msg.Attachments() {
+			_, url := handlers.SplitAttachment(a)
+			mediaURLs = append(mediaURLs, url)
+		}
+	} else {
+		text = handlers.GetTextAndAttachments(msg)
+	}
+
+	payload := mtPayload{From: msg.Channel().Address(), To: msg.URN().Path(), Body: text}
+	if len(mediaURLs) > 0 {
+		payload.MediaURL = strings.Join(mediaURLs, ",")
+	}
+
+	jsonPayload, err := json.Marshal(payload)
```

Comment:
I think it's ok to use `jsonx.MustMarshal` in cases like this since we control everything in that payload and we know it can be marshalled